diff '--color=auto' -ru libffado-2.5.0/SConstruct libffado-2.5.0-patched/SConstruct
--- libffado-2.5.0/SConstruct	2026-01-06 10:31:32.000000000 +0000
+++ libffado-2.5.0-patched/SConstruct	2026-06-07 08:33:06.418168764 +0100
@@ -284,9 +284,9 @@
         pkgs['libavc1394'] = '0.5.3'
 
     if not env['SERIALIZE_USE_EXPAT']:
-        if conf.CheckPKG('libxml++-3.0'):
-            pkgs['libxml++-3.0'] = '3.0.0'
-        if not('libxml++-3.0' in pkgs):
+        if conf.CheckPKG('libxml++-5.0'):
+            pkgs['libxml++-5.0'] = '5.0.0'
+        if not('libxml++-5.0' in pkgs):
             pkgs['libxml++-2.6'] = '2.13.0'
 
     # Provide a way for users to compile newer libffado which will work
@@ -369,7 +369,7 @@
     # produced.  Add the necessary option to CXXFLAGS if required.
     if conf.CheckPKG('libxml++-2.6 >= 2.39.1'):
         env.Append(CXXFLAGS = '-std=gnu++11')
-    if conf.CheckPKG('libxml++-3.0 >= 3.0.0'):
+    if conf.CheckPKG('libxml++-5.0 >= 5.0.0'):
         env.Append(CXXFLAGS = '-std=gnu++11')
 
     # Check for C99 lrint() and lrintf() functions used to convert from
diff '--color=auto' -ru libffado-2.5.0/src/libutil/serialize_libxml.cpp libffado-2.5.0-patched/src/libutil/serialize_libxml.cpp
--- libffado-2.5.0/src/libutil/serialize_libxml.cpp	2021-05-17 01:45:29.000000000 +0100
+++ libffado-2.5.0-patched/src/libutil/serialize_libxml.cpp	2026-06-07 08:39:16.298318321 +0100
@@ -75,7 +75,7 @@
 void
 Util::XMLSerialize::writeVersion()
 {
-#if LIBXMLXX_MAJOR_VERSION == 3
+#if LIBXMLXX_MAJOR_VERSION == 5
     xmlpp::Element* pElem =  m_doc.get_root_node()->add_child_element( "CacheVersion" );
 #else
     xmlpp::Node* pNode = m_doc.get_root_node();
@@ -83,7 +83,7 @@
 #endif
     char* valstr;
     asprintf( &valstr, "%s", CACHE_VERSION );
-#if LIBXMLXX_MAJOR_VERSION == 3
+#if LIBXMLXX_MAJOR_VERSION == 5
     pElem->set_first_child_text( valstr );
 #else
     pElem->set_child_text( valstr );
@@ -111,14 +111,14 @@
     pNode = getNodePath( pNode, tokens );
 
     // element to be added
-#if LIBXMLXX_MAJOR_VERSION == 3
+#if LIBXMLXX_MAJOR_VERSION == 5
     xmlpp::Element* pElem = pNode->add_child_element( tokens[tokens.size() - 1] );
 #else
     xmlpp::Element* pElem = pNode->add_child( tokens[tokens.size() - 1] );
 #endif
     char* valstr;
     asprintf( &valstr, "%lld", value );
-#if LIBXMLXX_MAJOR_VERSION == 3
+#if LIBXMLXX_MAJOR_VERSION == 5
     pElem->set_first_child_text( valstr );
 #else
     pElem->set_child_text( valstr );
@@ -147,7 +147,7 @@
     pNode = getNodePath( pNode, tokens );
 
     // element to be added
-#if LIBXMLXX_MAJOR_VERSION == 3
+#if LIBXMLXX_MAJOR_VERSION == 5
     xmlpp::Element* pElem = pNode->add_child_element( tokens[tokens.size() - 1] );
     pElem->set_first_child_text( str );
 #else
@@ -192,7 +192,7 @@
     }
 
     for ( unsigned int i = iTokenIdx; i < tokens.size() - 1; i++, iTokenIdx++ ) {
-#if LIBXMLXX_MAJOR_VERSION == 3
+#if LIBXMLXX_MAJOR_VERSION == 5
         pCurNode = pCurNode->add_child_element( tokens[iTokenIdx] );
 #else
         pCurNode = pCurNode->add_child( tokens[iTokenIdx] );
@@ -251,7 +251,7 @@
 {
     std::string savedVersion;
     if (read( "CacheVersion", savedVersion )) {
-        Glib::ustring expectedVersion = CACHE_VERSION;
+        std::string expectedVersion = CACHE_VERSION;
         debugOutput( DEBUG_LEVEL_NORMAL, "Cache version: %s, expected: %s.\n", savedVersion.c_str(), expectedVersion.c_str() );
         if (expectedVersion == savedVersion) {
             debugOutput( DEBUG_LEVEL_VERBOSE, "Cache version OK.\n" );
@@ -279,7 +279,7 @@
 
     debugOutput( DEBUG_LEVEL_VERY_VERBOSE, "pNode = %s\n", pNode->get_name().c_str() );
 
-#if LIBXMLXX_MAJOR_VERSION == 3
+#if LIBXMLXX_MAJOR_VERSION == 5
     xmlpp::Node::NodeSet nodeSet = pNode->find( strMemberName );
     for ( xmlpp::Node::NodeSet::iterator it = nodeSet.begin();
           it != nodeSet.end();
@@ -295,7 +295,7 @@
             dynamic_cast< const xmlpp::Element* >( *it );
         if ( pElement && pElement->has_child_text() ) {
             char* tail;
-#if LIBXMLXX_MAJOR_VERSION == 3
+#if LIBXMLXX_MAJOR_VERSION == 5
             value = strtoll( pElement->get_first_child_text()->get_content().c_str(),
                              &tail, 0 );
 #else
@@ -327,7 +327,7 @@
     }
     xmlpp::Node* pNode = pDoc->get_root_node();
 
-#if LIBXMLXX_MAJOR_VERSION == 3
+#if LIBXMLXX_MAJOR_VERSION == 5
     xmlpp::Node::NodeSet nodeSet = pNode->find( strMemberName );
     for ( xmlpp::Node::NodeSet::iterator it = nodeSet.begin();
           it != nodeSet.end();
@@ -342,7 +342,7 @@
         const xmlpp::Element* pElement = dynamic_cast< const xmlpp::Element* >( *it );
         if ( pElement ) {
             if ( pElement->has_child_text() ) {
-#if LIBXMLXX_MAJOR_VERSION == 3
+#if LIBXMLXX_MAJOR_VERSION == 5
                 str = pElement->get_first_child_text()->get_content();
 #else
                 str = pElement->get_child_text()->get_content();
@@ -370,7 +370,7 @@
         return false;
     }
     xmlpp::Node* pNode = pDoc->get_root_node();
-#if LIBXMLXX_MAJOR_VERSION == 3
+#if LIBXMLXX_MAJOR_VERSION == 5
     xmlpp::Node::NodeSet nodeSet = pNode->find( strMemberName );
 #else
     xmlpp::NodeSet nodeSet = pNode->find( strMemberName );
diff '--color=auto' -ru libffado-2.5.0/src/SConscript libffado-2.5.0-patched/src/SConscript
--- libffado-2.5.0/src/SConscript	2021-05-17 01:45:29.000000000 +0100
+++ libffado-2.5.0-patched/src/SConscript	2026-06-07 08:32:45.757756538 +0100
@@ -288,9 +288,9 @@
     libenv.MergeFlags( env['LIBIEC61883_FLAGS'].decode() )
     libenv.MergeFlags( env['LIBCONFIG_FLAGS'].decode() )
     if not env['SERIALIZE_USE_EXPAT']:
-        if 'LIBXML30_FLAGS' in env :
-            libenv.MergeFlags( env['LIBXML30_FLAGS'].decode() )
-        if not('LIBXML30_FLAGS' in env) :
+        if 'LIBXML50_FLAGS' in env :
+            libenv.MergeFlags( env['LIBXML50_FLAGS'].decode() )
+        if not('LIBXML50_FLAGS' in env) :
             libenv.MergeFlags( env['LIBXML26_FLAGS'].decode() )
     else:
         libenv.PrependUnique( LIBS=["expat"] )
diff '--color=auto' -ru libffado-2.5.0/support/dbus/SConscript libffado-2.5.0-patched/support/dbus/SConscript
--- libffado-2.5.0/support/dbus/SConscript	2021-05-17 01:45:29.000000000 +0100
+++ libffado-2.5.0-patched/support/dbus/SConscript	2026-06-07 08:34:32.315588015 +0100
@@ -41,9 +41,9 @@
     env.MergeFlags( env["DBUSC1_FLAGS"].decode() )
     env.MergeFlags( env['LIBRAW1394_FLAGS'].decode() )
     if not env['SERIALIZE_USE_EXPAT']:
-        if 'LIBXML30_FLAGS' in env :
-            env.MergeFlags( env['LIBXML30_FLAGS'].decode() )
-        if not('LIBXML30_FLAGS' in env) :
+        if 'LIBXML50_FLAGS' in env :
+            env.MergeFlags( env['LIBXML50_FLAGS'].decode() )
+        if not('LIBXML50_FLAGS' in env) :
             env.MergeFlags( env['LIBXML26_FLAGS'].decode() )
     else:
         env.PrependUnique( LIBS=["expat"] )
diff '--color=auto' -ru libffado-2.5.0/support/firmware/SConscript libffado-2.5.0-patched/support/firmware/SConscript
--- libffado-2.5.0/support/firmware/SConscript	2021-05-17 01:45:29.000000000 +0100
+++ libffado-2.5.0-patched/support/firmware/SConscript	2026-06-07 08:34:13.316442045 +0100
@@ -33,9 +33,9 @@
     env.MergeFlags( env['LIBRAW1394_FLAGS'].decode() )
     env.MergeFlags( env['LIBIEC61883_FLAGS'].decode() )
     if not env['SERIALIZE_USE_EXPAT']:
-        if 'LIBXML30_FLAGS' in env :
-            env.MergeFlags( env['LIBXML30_FLAGS'].decode() )
-        if not('LIBXML30_FLAGS' in env) :
+        if 'LIBXML50_FLAGS' in env :
+            env.MergeFlags( env['LIBXML50_FLAGS'].decode() )
+        if not('LIBXML50_FLAGS' in env) :
             env.MergeFlags( env['LIBXML26_FLAGS'].decode() )
     else:
         env.PrependUnique( LIBS=["expat"] )
diff '--color=auto' -ru libffado-2.5.0/support/tools/SConscript libffado-2.5.0-patched/support/tools/SConscript
--- libffado-2.5.0/support/tools/SConscript	2021-05-17 01:45:29.000000000 +0100
+++ libffado-2.5.0-patched/support/tools/SConscript	2026-06-07 08:33:55.245193139 +0100
@@ -33,9 +33,9 @@
 e.MergeFlags( "-I#/ -I#/src -L%ssrc -lffado" % env['build_base'] )
 if not e.GetOption( "clean" ):
     if not env['SERIALIZE_USE_EXPAT']:
-        if 'LIBXML30_FLAGS' in env :
-            e.MergeFlags( env['LIBXML30_FLAGS'].decode() )
-        if not('LIBXML30_FLAGS' in env) :
+        if 'LIBXML50_FLAGS' in env :
+            e.MergeFlags( env['LIBXML50_FLAGS'].decode() )
+        if not('LIBXML50_FLAGS' in env) :
             e.MergeFlags( env['LIBXML26_FLAGS'].decode() )
     else:
         e.PrependUnique( LIBS=["expat"] )
diff '--color=auto' -ru libffado-2.5.0/tests/SConscript libffado-2.5.0-patched/tests/SConscript
--- libffado-2.5.0/tests/SConscript	2021-05-17 01:45:29.000000000 +0100
+++ libffado-2.5.0-patched/tests/SConscript	2026-06-07 08:34:46.536972003 +0100
@@ -32,9 +32,9 @@
     env.MergeFlags( env['LIBIEC61883_FLAGS'].decode() )
     env.MergeFlags( env['LIBRAW1394_FLAGS'].decode() )
     if not env['SERIALIZE_USE_EXPAT']:
-        if 'LIBXML30_FLAGS' in env :
-             env.MergeFlags( env['LIBXML30_FLAGS'].decode() )
-        if not('LIBXML30_FLAGS' in env) :
+        if 'LIBXML50_FLAGS' in env :
+             env.MergeFlags( env['LIBXML50_FLAGS'].decode() )
+        if not('LIBXML50_FLAGS' in env) :
              env.MergeFlags( env['LIBXML26_FLAGS'].decode() )
     else:
         env.PrependUnique( LIBS=["expat"] )
